### PR TITLE
Add more linters

### PR DESCRIPTION
**Pull request by lucacome**
### Proposed changes

Adds more linters and fixes errors

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-asg-sync/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: cmd/sync/config.go [hunk 1]
```diff
@@ -8,18 +8,18 @@ import (
...
 
-// commonConfig stores the configuration parameters common to all providers
+// commonConfig stores the configuration parameters common to all providers.
 type commonConfig struct {
-	APIEndpoint           string        `yaml:"api_endpoint"`
-	CloudProvider         string        `yaml:"cloud_provider"`
-	SyncIntervalInSeconds time.Duration `yaml:"sync_interval_in_seconds"`
+	APIEndpoint   string        `yaml:"api_endpoint"`
+	CloudProvider string        `yaml:"cloud_provider"`
+	SyncInterval  time.Duration `yaml:"sync_interval_in_seconds"`
```
Comment (kate-osborn) [inline cmd/sync/config.go:15]: Why this change?

Comment (lucacome) [reply cmd/sync/config.go:15]:
```
ST1011: var SyncIntervalInSeconds is of type time.Duration; don't use unit-specific suffix "Seconds" (stylecheck)
```

Comment (lucacome) [reply cmd/sync/config.go:15]: @kate-osborn 👀

**Review by kate-osborn — APPROVED** [review]
